sdk/nodejs/docker.ts

```diff
@@ -116,13 +115,14 @@ function useDockerPasswordStdin(logResource: pulumi.Resource) {
  */
 export function buildAndPushImage(
     imageName: string,
-    pathOrBuild: string | DockerBuild,
+    pathOrBuild: pulumi.Input<string | DockerBuild>,
     repositoryUrl: pulumi.Input<string>,
     logResource: pulumi.Resource,
-    connectToRegistry: () => Promise<Registry>): pulumi.Output<string> {
+    connectToRegistry: () => pulumi.Input<Registry>): pulumi.Output<string> {
 
-    return pulumi.output(repositoryUrl).apply(repoUrl =>
-        buildAndPushImageAsync(imageName, pathOrBuild, repoUrl, logResource, connectToRegistry));
+    return pulumi.all([pathOrBuild, repositoryUrl])
```

Comment:
Not sure I follow this - why are we doing an apply here if we're going to ignore the apply arguments and use the inputs directly? Can't we just call `buildAndPushImageAsync`?

Reply:
will see that in the next PR ;0